appinventor/blocklyeditor/src/component_database.js

```diff
@@ -360,6 +415,79 @@ Blockly.ComponentDatabase.prototype.populateTypes = function(componentInfos) {
   }
 };
 
+/**
+ * Processes the given array of parameters (from simple_components.json, not
+ * Parameters) and returns an array of Parameters.
+ * @param {!Array.<!Object>} paramData An array of data from
+ *     simple_components.json defining the parameters.
+ * @return {!Array.<!Parameter>} An array of parameters.
+ */
+Blockly.ComponentDatabase.prototype.processParameters = function(paramData) {
+  params = [];
+  for (var i = 0, datum; datum = paramData[i]; i++) {
+    var param = {};
+    param.name = datum.name;
+    param.type = datum.type;
+    param.helperKey = this.processHelper(datum.helper);
+    params.push(param);
+  }
+  return params;
+}
+
+/**
+ * Processes a helper (from simple_components.json) and returns a HelperKey if
+ * possible.
+ * @param {Object} helper The (possibly null) helper definition.
+ * @return {HelperKey} The helper key associated with the helper (if it is
+ *     possible) to create one.
+ */
+Blockly.ComponentDatabase.prototype.processHelper = function(helper) {
+  if (!helper) {
+    return null;
+  }
+  switch (helper.type) {
+    case "OPTION_LIST":
+      return this.processOptionList(helper.data);
+  }
+  return null;
+}
+
+/**
+ * Processes data defining an OptionList (from simple_components.json) and
+ * returns a HelperKey associated with the OptionList.
```

Comment:
Should say here that this method is used in conjunction with the optionLists_ field to de-duplicated duplicated optionList information in simple-components.json.

Reply:
[Done =)](https://github.com/BeksOmega/appinventor-sources/pull/4/files#diff-7f92aaa2f89110a614c406da22dc510cR466)